src/Common/Menus/Traits/With_Settings.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Provides methods and properties for submenus.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus\Traits;
+
+trait With_Settings {
+
+	/**
+	 * path to the settings file.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $settings_file = '';
+
+	/**
+	 * Constructor
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $data The data for creating the settings page.
+	 */
+	public function init( array $data ) {
```

Comment:
Too generic a method name for a trait.

Reply:
Furthermore, this looks like something that should live in a `__construct` method, not in a method that implementers have to remember to call.